baselines/flwr_baselines/publications/leaf/femnist/test_utils.py

```diff
@@ -0,0 +1,53 @@
+"""Utils module for tests."""
```

Comment:
This must be moved to `dataset/utils.py` as it only relates to the dataset tests and doesn't need to stand in another file as it also contains utility functions.

Reply:
This file is not the utils for the code but only for the test. I don't think it should be moved there

Reply:
I know, but it still provides utils for members of the directory (and only for them). It wouldn't make sense inside a specific directory to have multiple utils files for different modules. If it does makes more sense, then it is a sign that new folders must be created, but in our case, we want the tests to seat beside the files so the `dataset/utils.py` will hold the utils functions for all files in the `dataset/` folder.

Reply:
I don't agree. The reason for having a module is some separation and division by functionality, even with the directory structure that we have. And even if the names are a little bit similar (that doesn't matter). Think about a potential first-time user. Which version will he find more confusing? Finding some random function merge into a single file or having a separate file that holds functions that were created for a completely different purpose.

Reply:
But this is the purpose of a `utils.py`, to merge a few functions that are used in multiple files without necessarily a common theme, otherwise we would create specific modules instead of a broad `utils.py`. 
I don't disagree with you on this, I just thought it would be cleaner to have everything in one file while still making sense in my opinion.

Reply:
Our general approach is to have tests right next to the module they test. We always follow the pattern of having `some_module.py` / `some_module_test.py` in the same directory. This makes it easy to find and edit the tests for a module. and our build tools will then ignore `_test.py`files when building the wheel.

One thing we didn't run into yet is a situation where we need test utils. Because we don't have a separate test directory, a `test_utils` would be sitting there all alone (and the build tools would potentially package it with a release). Moving the test utils into a `utils` module is also not the best option, because that would turn test code into application/framework code.

My suggestion would be to move the test utilities into one of the other test modules (e.g., at the end or beginning of `some_module_test.py`), and then import those utilities in other test modules (`from some_module_test import my_util_fn`).

Reply:
I moved the functions from test_utils.py to dataset_test.py. It seemed the most appropriate.